src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -96,6 +99,49 @@ public TopicController(ValidatingTopicService topicService, KafkaClusters kafkaE
         }
     }
 
+    @GetMapping(value = "/api/applicationsmetadata/{envId}/{topicName}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<KnownApplicationDto> getAvailableApplicationMetadata(@PathVariable String envId,
+            @PathVariable String topicName,
+            @RequestParam(name = "topicOwnerId", required = false) String topicOwnerId) {
+        TopicMetadata topic = topicService.getTopic(envId, topicName).orElseThrow(notFound);
+        List<String> producerIds = topic.getProducers();
+
+        return applicationsService.getAllApplicationMetadata(envId).stream().map(ApplicationMetadata::getApplicationId)
+                .filter(id -> !id.equals(topicOwnerId))
+                .filter(idsWithoutOwnerid -> !producerIds.contains(idsWithoutOwnerid))
+                .map(id -> toKnownAppDto(applicationsService.getKnownApplication(id)
+                        .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND))))
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping(value = "/api/producers/{environmentId}", consumes = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
The topic name should be in the endpoint path, not in the DTO! Recommend `/api/producers/{environmentId}/{topicName}`.